CodeStarterCamp_Week3/main.swift

```diff
@@ -9,36 +9,76 @@
 import Foundation
 
 class Person {
-    var money: Int = 0
+    var name: String
+    var money: Int
     
-    func buyCoffee() {
-        
+    init(name: String, money: Int) {
+        self.name = name
+        self.money = money
+    }
+    
+    func buyCoffee(coffeeShop: CoffeeShop, nameOfCoffee: Coffee) {
+        if let price = coffeeShop.menu[nameOfCoffee], self.money >= price {
+            self.money -= price
+            print("\(nameOfCoffee)를 구입 하였습니다.")
+            coffeeShop.takeOrder(nameOfCoffee: nameOfCoffee)
+            coffeeShop.brew(coffee: nameOfCoffee)
+        } else {
+            print("돈이 부족하여 구입할 수 없습니다.")
+        }
     }
 }
 
 class CoffeeShop {
     var sales: Int = 0
-    var menu: [String: Int] = [:]
-    var pickUpTable: [String] = []
-    var barista: Person?
+    var menu: [Coffee: Int] = [:]
+    var pickUpTable: [Coffee] = []
+    var barista: Person
     
-    enum Coffee: String {
-        case espresso, americano, cafeLatte, vanilaLatte, cafeMoca
+    init(menu: [Coffee: Int], barista: Person) {
+        self.menu = menu
+        self.barista = barista
     }
     
-    func getOrder(orderedCoffee: String) {
-        print("\(orderedCoffee)를 주문 받았다.")
-        let brewedCoffee: String = brewCoffee(to: orderedCoffee)
-        pickUpTable.append(brewedCoffee)
+    func takeOrder(nameOfCoffee: Coffee) {
+        print("\(nameOfCoffee)를 주문 받았다.")
+        if let income = self.menu[nameOfCoffee] {
+            self.sales += income
+            print("매출이 \(income)원 증가하였습니다.")
+        }
     }
     
-    func brewCoffee(to orderedCoffee: String) -> String {
-        let brewedCoffee: String = "완성된 \(orderedCoffee)"
-        return brewedCoffee
+    func brew(coffee: Coffee) {
+        print("\(coffee)를 완성했다.")
+        pickUpTable.append(coffee)
     }
 }
 
-var misterLee: Person = Person()
-var yagombucks: CoffeeShop = CoffeeShop()
-yagombucks.barista = misterLee
+enum Coffee: String {
+    case espresso
+    case americano
+    case cafeLatte
+    case vanilaLatte
+    
+    var price: Int {
+        get {
+            switch self {
+            case .espresso: return 2500
+            case .americano: return 3500
+            case .cafeLatte: return 4000
+            case .vanilaLatte: return 4500
+            }
+        }
+    }
+}
+
+var yagombucksMenu: [Coffee: Int] = [.americano: Coffee.americano.price, .espresso: Coffee.espresso.price, .cafeLatte: Coffee.cafeLatte.price, .vanilaLatte: Coffee.vanilaLatte.price]
```

Comment:
한 줄에 코드가 너무 길 경우에는 줄바꿈해주는 컨벤션도 있답니다 !
```suggestion
var yagombucksMenu: [Coffee: Int] = [
    .americano: Coffee.americano.price, 
    .espresso: Coffee.espresso.price, 
    .cafeLatte: Coffee.cafeLatte.price, 
    .vanilaLatte: Coffee.vanilaLatte.price
]
```